### PR TITLE
feat(patients): max DOB set for current date

### LINKED ISSUE #1984: Date of Birth date picker should have max date of the current date
## 🚀 Feature Proposal

The Date Picker component supports defining a maximum date. The date of birth date picker should set the maximum date as today. 

## Motivation

Ease of use

### PR DESCRIPTION
**Pull request by jlato89**
Added maxDate prop to DatePicker component to avoid DOB being set in the future.

Fixes #1984.

**Changes proposed in this pull request:**

- Added maxDate prop to DatePickerWithLabelFormGroup component to restrict DOB from being set in the future.

**Newly added dependencies with [Bundlephobia](https://bundlephobia.com/) links:**

- No new dependencies

### CODE REVIEW

**Comment by jlato89** [comment]
This also Fixes #1976 regarding the date of birth

File: src/components/input/DatePickerWithLabelFormGroup.tsx [hunk 1]
```diff
@@ -22,6 +23,7 @@ const DatePickerWithLabelFormGroup = (props: Props) => {
         dateFormat="MM/dd/yyyy"
         dateFormatCalendar="LLLL yyyy"
         dropdownMode="scroll"
+        maxDate={addDays(new Date(), 0)}
```
Comment (jackcmeyer) [inline src/components/input/DatePickerWithLabelFormGroup.tsx:26]: This component is intended to be more general than for use inside of the new/edit patient screen. 

I think that this interface will only need to expose an optional maxDate and just pass it along to the `DateTimePicker from `@hopsitalrun/components`. 

Then we'll need to make a change in the `GenrealInformationComponent` to set the `maxDate` property.

Comment (fox1t) [reply src/components/input/DatePickerWithLabelFormGroup.tsx:26]: I agree 100% with your vision. @jlato89 can you handle it?

**Comment by jlato89** [comment]
I had a feeling it was but couldn't find another instance of it being used. 
Will work on those changes. Tests might take a bit longer as I havnt done 
them before but I'm sure il figure it out.

File: src/components/input/DatePickerWithLabelFormGroup.tsx (hunk 1 above)
Comment (jlato89) [reply src/components/input/DatePickerWithLabelFormGroup.tsx:26]: I should be able to yes. just an oversight on my part.

**Comment by jlato89** [comment]
Added the appropriate changes. Just have to figure out how todo the tests. Then it will be ready to be merged.

**Comment by jackcmeyer** [comment]
closing in favor of #2002

**Pull request closed without merging**